start-here.mdx

```diff
@@ -1,8 +1,153 @@
+
+Incoming external messages come from an external address, and outgoing external messages go to an external address.
+
+The _state_ of the account changes only when it handles messages. Messages also change the account's balance. To have a state and a balance, and to handle messages, the account must be `active`.
+
+Formally, a message is only an intent: it has a destination, possibly some Toncoin, and data. After the message is handled and all the necessary changes are made to the blockchain, the message is packed, along with a description of those changes, into a single packet of data, [transaction](/foundations/messages/ordinary-tx).
+
+A message might also be a [deploy message](/foundations/messages/deploy) if it has contract code and its initial state (`StateInit`) attached. When such a message is sent, the hash of the code and initial state must match the destination address. If they do, the code and state are stored in the account at that address, and the account becomes active. Both the code and state stored in the account may change in the future, but its address will remain the same as when it was originally deployed.
+
+Internal and incoming external messages execute the account's code. The code is interpreted by [TON Virtual Machine](/tvm/overview) (TVM). It is written in _bitcode_, a binary format specific to TVM. In the future, TVM might support multiple binary languages, _codepages_, but at the moment there is only codepage 0 (`CP0`).
+
+Most development is done in the [Tolk](/languages/tolk/overview), a high-level programming language. Its compiler is included in the [Blueprint](/contract-dev/blueprint/overview) development environment and will be available to all projects created from the template.
+
+Originally, [Fift](/languages/fift/overview), a Forth-like assembly language, and [FunC](/languages/func/overview), a C-like intermediate-level language, were used for contract development.
+
+When execution [starts](/tvm/initialization), the message and current account state are provided to the code. By the end of execution, the destination account might change its state or code, or send internal or outgoing external messages.
+
+The execution follows a procedure whose steps are called [phases](/foundations/phases). [Fees](/foundations/fees) are deducted during this process. Fees might be deducted from the account's balance or from the message, depending on the [mode](/foundations/messages/modes) of the message.
+
+Execution cost is first measured in _gas_ units, then converted to Toncoin. This unit is separate so that if code execution becomes computationally cheaper (or more expensive), validators can vote to change the [price of gas](/foundations/config#param-20-and-21:-gas-prices) in Toncoin.
+
+If something goes wrong, a non-zero [exit code](/tvm/exit-codes) might be returned, no changes to state or code are saved, and no further messages are sent.
```

Comment:
there's not a single mention of bounces on this page... and i think in this specific place it would be a nice thing to mention because it's very important!